Celeste.Mod.mm/Mod/Helpers/SwapImmediatelyExtension.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Monocle;
+using System;
+using System.Collections;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Celeste.Mod {
+    public static class SwapImmediatelyExtension {
+        /// <summary>
+        ///   when you want to enumerate a coroutine ienumerator,
+        ///   always keep in mind that the original sequence may have been wrapped into <see cref="SwapImmediately"/>.
+        ///   this method helps you to handle it correctly like <see cref="Coroutine"/>.<br/>
+        ///   also known as, flattening the given sequence.
+        /// </summary>
+        /// <returns>
+        ///   A new enumerator, but all <see cref="SwapImmediately"/> is safely handled.
+        /// </returns>
```

Comment:
```suggestion
        /// <summary>
        /// Flattens all <see cref="SwapImmediately"/> returned by the source IEnumerator, 
        /// that is, when the source IEnumerator would yield a <see cref="SwapImmediately"/>,
        /// elements from the wrapped IEnumerator get yielded one-by-one instead. 
        /// (Behaviour mimicks <see cref="Coroutine"/>'s handling of <see cref="SwapImmediately"/>.)
        /// When writing hooks on Coroutine-style methods, 
        /// this method is needed if you want to access values returned from the orig Coroutine,
        /// or rely on the source Coroutine yielding a specific amount of values.
        /// </summary>
        /// <returns>
        ///   A new enumerator, with all <see cref="SwapImmediately"/> flattened.
        /// </returns>
```

This should be a bit better imo, assuming I understand the reasoning behind this method existing.